authprox/oidc.go

```diff
@@ -0,0 +1,64 @@
+package authprox
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/coreos/go-oidc"
+)
+
+type oidcValidator struct {
+	sync.Mutex
+	issuers []*issuer
+}
+
+type issuer struct {
+	Issuer string
+	p      *oidc.Provider
+	cfg    oidc.Config
+	v      *oidc.IDTokenVerifier
+	ctx    context.Context
+}
+
+func (o *oidcValidator) FindClaim(issuerStr string, input []byte) (claim, extraData string, err error) {
+	o.Lock()
+	defer o.Unlock()
+
+	// find existing, or create a new issuer
+	var is *issuer
+	for i := range o.issuers {
+		if o.issuers[i].Issuer == issuerStr {
+			is = o.issuers[i]
+		}
+	}
+	if is == nil {
+		is = &issuer{
+			Issuer: issuerStr,
+			ctx:    context.Background(),
+		}
+		is.p, err = oidc.NewProvider(is.ctx, is.Issuer)
+		if err != nil {
+			return
+		}
+		is.cfg.SkipClientIDCheck = true
+		is.v = is.p.Verifier(&is.cfg)
+		o.issuers = append(o.issuers, is)
+	}
+
+	idToken, err := is.v.Verify(is.ctx, string(input))
```

Comment:
I was reading the documentation of the `Verify` function, it says "Verify does NOT do nonce validation, which is the callers responsibility". I don't know where the nonce is stored or what exactly needs to be verified, but are we doing it here? If it's not needed, could you add a comment to explain why?

Reply:
I've updated the docs a bit. This is also discussed in el/main.go, at the call to AuthCodeURL.